### PR TITLE
ADEN-4720 Support multiple video trigger areas

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [uapVideo.js#L14](https://github.com/Wikia/app/blob/58c09be18555a99ed026def49fd3bce03345d16a/extensions/wikia/AdEngine/js/video/uapVideo.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ define('ext.wikia.adEngine.template.bfaaMobile', [
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
 ], function (uapContext,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js:12]: MAJOR Function has 9 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [uapVideo.js#L14](https://github.com/Wikia/app/blob/3e26bf0960cc33c89757a6e19c68c44bf561adc4/extensions/wikia/AdEngine/js/video/uapVideo.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 2]
```diff
@@ -12,8 +12,8 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
 ], function (uapContext,
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js:13]: MAJOR Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [uapVideo.js#L14](https://github.com/Wikia/app/blob/c748ba7329c90bd2e038ed9b02885edbf746648b/extensions/wikia/AdEngine/js/video/uapVideo.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [uapVideo.js#L14](https://github.com/Wikia/app/blob/f767b216d089e755d2a0d44fde64116a315aab5c/extensions/wikia/AdEngine/js/video/uapVideo.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [uapVideo.js#L14](https://github.com/Wikia/app/blob/08b108b91bf357b2c8c1e203cdb0ab86163d8e48/extensions/wikia/AdEngine/js/video/uapVideo.js#L14): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/video/uapVideo.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ define('ext.wikia.adEngine.video.uapVideo', [
 		// On mercury splitLayoutVideoPosition and videoPlaceholderElement will be empty
 		// because we always display video in the same way there.
 		if (params.splitLayoutVideoPosition && params.videoPlaceholderElement) {
-			width = params.videoPlaceholderElement.width;
+			width = params.videoPlaceholderElement.offsetWidth;
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/uapVideo.js:28]: This small change removes sometimes a perception of a gap between first and second part of the ad, right?

File: extensions/wikia/AdEngine/js/slot/resolvedState.js [hunk 4]
```diff
@@ -26,20 +26,12 @@ define('ext.wikia.adEngine.slot.resolvedState', [
...
-		return correctSingleImage !== '' || correctMultipleImages;
-	}
-
 	function createCacheKey() {
 		return cacheKey + '_' + uapContext.getUapId();
 	}
 
 	function findRecordInCache() {
-		return cache.get(createCacheKey(), now)
+		return cache.get(createCacheKey(), now);
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/slot/resolvedState.js:34]: Eagle-eye! :eagle:

File: extensions/wikia/AdEngine/js/slot/resolvedState.js [hunk 5]
```diff
@@ -80,29 +72,24 @@ define('ext.wikia.adEngine.slot.resolvedState', [
...
 	function setResolvedState(params) {
 		log('Resolved state is turned on', logGroup, log.levels.debug);
-		if (params.backgroundImage) {
-			params.aspectRatio = params.resolvedState.aspectRatio;
-			params.backgroundImage.src = params.resolvedState.imageSrc;
-		} else {
-			params.aspectRatio = params.resolvedStateAspectRatio;
-			params.image1.element.src = params.image1.resolvedStateSrc;
+		params.aspectRatio = params.resolvedStateAspectRatio;
+		params.image1.element.src = params.image1.resolvedStateSrc;
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/slot/resolvedState.js:76]: ~~Is `image1` required in DFP creative template?~~ It is :) I was able to check it by my own :japanese_ogre:

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**